packages/core/src/app/payment/paymentMethod/AdyenV2CardValidation.tsx

```diff
@@ -1,62 +1,94 @@
+    paymentMethod: PaymentMethod;
+    cardValidationState?: AdyenV2ValidationState;
+    selectedInstrument?: CardInstrument;
 }
 
 const AdyenV2CardValidation: FunctionComponent<AdyenV2CardValidationProps> = ({
     verificationFieldsContainerId,
     shouldShowNumberField,
-    paymentMethodType,
+    selectedInstrument,
+    paymentMethod,
     cardValidationState,
 }) => {
-    const fieldsValidation: FieldsValidation = useMemo(() => {
-        const newFieldValidation: FieldsValidation = {};
+    const paymentContext = useContext(PaymentContext);
+
+    const shouldShowSecurityCode = paymentMethod.method === 'scheme';
+    const shoudlShowExpiryDate = paymentMethod.method === 'bcmc';
+
+    const validationState: FieldsValidation = {}
+    if (shouldShowNumberField) {
+        validationState[AdyenV2CardFields.CardNumber] = { valid: false };
+    }
+    if (shouldShowSecurityCode) {
+        validationState[AdyenV2CardFields.SecurityCode] = { valid: false };
+    }
+    if (shoudlShowExpiryDate) {
+        validationState[AdyenV2CardFields.ExpiryDate] = { valid: false };
+    }
+
+    const [ fieldsValidation, setFieldsValidation ] = useState<FieldsValidation>(validationState);
+
+    useEffect(() => {
         if (!cardValidationState) {
-            return newFieldValidation;
+            return;
         }
 
-        if (Object.keys(cardValidationState).length === 0) {
-            Object.values(AdyenV2CardFields).forEach((key: AdyenV2CardFields) => {
-                newFieldValidation[key] = {valid: false} as AdyenV2CardValidationState;
-            });
-        } else if (cardValidationState.fieldType) {
-            newFieldValidation[cardValidationState.fieldType] = cardValidationState;
+
+        if (cardValidationState.fieldType) {
+            if(cardValidationState.fieldType === AdyenV2CardFields.CardNumber) {
+                setFieldsValidation({
+                    ...fieldsValidation,
+                    [AdyenV2CardFields.CardNumber]: cardValidationState?.endDigits !== selectedInstrument?.last4 ?
+                        { ...cardValidationState, valid: false } :
+                        { ...cardValidationState },
+                })
+            } else {
+                setFieldsValidation({
+                    ...fieldsValidation,
+                    [cardValidationState.fieldType]: cardValidationState,
+                })
+            }
         }
 
-        return newFieldValidation;
     }, [cardValidationState]);
 
-    const isFieldInvalid = (fieldKey: AdyenV2CardFields): boolean => {
-        return !!fieldsValidation[fieldKey] && !fieldsValidation[fieldKey]?.valid;
-    };
+    const isFieldInvalid = (fieldKey: AdyenV2CardFields): boolean => !!fieldsValidation[fieldKey] && !fieldsValidation[fieldKey]?.valid;
+
+    useEffect(() => {
+        paymentContext?.disableSubmit(paymentMethod, Object.values(fieldsValidation).some(field => !field?.valid));
+    }, [fieldsValidation]);
```

Comment:
Here may need to add paymentContext, paymentMethod and fieldsValidation? Also a general rule of thumb useEffect probably is better to be placed on top of the function?